### PR TITLE
Fix for plasma rifle objective not completing, more fixes

### PR DESCRIPTION
**Pull request by thesourcehim**
Can someone test this please? I no longer have a save file near there.
EDIT: Tested, it works, but it appears I need to compile map file as bsp (I edited save file manually to assign names). I'll need to recompile the map before merging.

### CODE REVIEW

**Comment by thesourcehim** [comment]
Ok, I recompiled the map, now objective completes as soon as I pick any of the 6 plasma rifles in the secret room (used q3map2 -bsp -onlyents)

**Comment by thesourcehim** [comment]
Slightly changed initialization. KWin blocks window resize beyond some desktop elements which results in wrong resolution after entering fullscreen.

File: src/init.cpp [hunk 1]
```diff
@@ -421,10 +421,6 @@ void initSystem()
 		SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 0);
 	}
 
-	SDL_GL_SetAttribute(SDL_GL_ACCUM_RED_SIZE, 16);
-	SDL_GL_SetAttribute(SDL_GL_ACCUM_GREEN_SIZE, 16);
-	SDL_GL_SetAttribute(SDL_GL_ACCUM_BLUE_SIZE, 16);
```
Comment (gsliepen) [inline src/init.cpp:426]: But this breaks motion blur support on X11. Perhaps it's better to test for accumulation buffer support, and if not supported don't request is and also don't call `updateScreenMB()` in cplusplus/CGraphics.cpp anymore.

Comment (thesourcehim) [reply src/init.cpp:426]: Agreed. I'll change this tomorrow.

**Comment by gsliepen** [comment]
Hi, the accumulation buffer changes look good. But the fullscreen resolution changing still has issues (probably existing before your changes though): if you are in fullscreen mode and change the resolution, that works fine, but then switching back to windowed mode results in the wrong window size.

Apart from that, can you split this pull request into three separate pull requests? One for the plasma rifle objective fixes, one for the fullscreen fixes, and one for the accumulation buffer? This way they can be reviewed and merged separately.

**Comment by thesourcehim** [comment]
> Hi, the accumulation buffer changes look good. But the fullscreen resolution changing still has issues (probably existing before your changes though): if you are in fullscreen mode and change the resolution, that works fine, but then switching back to windowed mode results in the wrong window size.
> 
> Apart from that, can you split this pull request into three separate pull requests? One for the plasma rifle objective fixes, one for the fullscreen fixes, and one for the accumulation buffer? This way they can be reviewed and merged separately.

Just noticed I forgot to initialize accum size variables.
Sure, I'll split this PR tomorrow.
UPDATE: turns out it's harder than I thought, those commits depend on each other. I guess I have to do them again manually in 3 different branches.

**Comment by thesourcehim** [comment]
I have added 2 separate pull requests. I'll add the one for accum buffers after fullscreen one is finished. I'm closing this one now.

**Pull request closed without merging**